livingatlas/pipelines/src/main/java/au/org/ala/pipelines/beam/ImageServiceDiffLoadPipeline.java

```diff
@@ -115,35 +115,40 @@ public void processElement(
                           @Element String imageMapping, OutputReceiver<KV<String, Multimedia>> out)
                           throws Exception {
 
-                        final CSVParser parser = new CSVParser();
-                        String[] parts = parser.parseLine(imageMapping);
-
-                        if (parts.length != NO_OF_CSV_FIELDS) {
-                          throw new RuntimeException("Problem with line: " + imageMapping);
+                        try {
+                          final CSVParser parser = new CSVParser();
+                          String[] parts = parser.parseLine(imageMapping);
+
+                          if (parts.length == NO_OF_CSV_FIELDS) {
+
+                            // CSV is imageID
+                            // Swap so we key on URL for later grouping
+                            Multimedia multimedia =
+                                Multimedia.newBuilder()
+                                    .setIdentifier(parts[0]) // image service ID
```

Comment:
More a general question. Is there a way of doing this that uses names rather than fragile positions?